src/Functions/toStartOfInterval.cpp

```diff
@@ -17,15 +20,297 @@ namespace DB
 {
 namespace ErrorCodes
 {
-    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ARGUMENT_OUT_OF_BOUND;
+    extern const int BAD_ARGUMENTS;
+    extern const int DECIMAL_OVERFLOW;
     extern const int ILLEGAL_COLUMN;
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
-    extern const int ARGUMENT_OUT_OF_BOUND;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
 }
 
 
 namespace
 {
+constexpr auto function_name = "toStartOfInterval";
```

Comment:
I am confused, why were l. 34 - 313 copypasted from DateTimeTransforms.h? Please check if there are further callers of the original code in DateTimeTransforms. If not, please move the code back and do all necessary code changes in DateTimeTransforms.